### ISSUE TITLE
Add support for RFC 5005 (http://podlove.org/paged-feeds/)

### ISSUE THREAD

**Issue by mase76** [gpodder/gpodder#231]
Hi!
I have problems with some feeds, for example
http://cre.fm/feed/m4a/
I guess, you know that feed. Older episodes than CRE154 are not parsed correctly.
Similar problems in
http://not-safe-for-work.de/feed/m4a/
In gpodder.net they appear as unknown episode. In the desktop client they don't
even appear. The Sailfish client fetches them correctly.

**Comment by thp** [comment]
Well, the first "page" of the feed only contains episodes from ~ CRE154 onwards. The gpodder-core (= 4.x, Sailfish OS) version has the handling for the paged feeds in `src/gpodder/plugins/podcast.py`, whereas this one (= 3.x, Desktop) doesn't include that support yet. I'm retitling the issue.

**Comment by jdelStrother** [comment]
What would need to happen to get this into the desktop client?

**Comment by cnleo** [comment]
We need the support for
`<link rel="next" href="http://example.org/index.atom?page=2"/>`
and
`<atom:link rel="next" href="http://example.org/index.atom?page=2"/>`
to get all the nice podcasts episodes out there. 

https://www.ibm.com/developerworks/library/x-tipatom2/
https://tools.ietf.org/html/rfc5005#section-3

**Comment by chlorophyll-zz** [comment]
we have a pr #184 which looks mostly the same but for soundcloud feeds.

**Comment by elelay** [comment]
This is finally implemented.
Please test and give me feedback if it doesn't work for you :-)

**Comment by chlorophyll-zz** [comment]
tested with http://cre.fm/feed/m4a/ and works well. thank you for your contribution.

**Comment by cnleo** [comment]
Does not work on my site.

gPodder 3.10.9 (2019-06-09) Windows 7

I have problems to import the both urls from @chlorophyll-zz too.
Import works with https://feeds.metaebene.me/cre/m4a but no paganation (only episode 160 to 219)

All podcasts on my site do not load the paged feeds.
Settings are by 1000 episodes per podcast.

Some Ideas, windows issue, config thing on my site?

**Comment by chlorophyll-zz** [comment]
@cnleo the slash at the end do not matter much. 
But the correct feed urls are in fact http://cre.fm/feed/m4a and http://not-safe-for-work.de/feed/m4a . 

Have you tried cleaning the gpodder folder inside your user folder? 

If that does not work, could you follow these debugging instructions and quote the log? Then the Developers could track your specific bug and fix it for everyone affected. (In short: "gpodder.exe -v")

From: https://gpodder.github.io/docs/windows.html
Debugging gPodder on Windows
Both the portable and installer version of gPodder come with a console version for debug messages. To launch it:

Choose Start > Run… or use the Win-R shortcut
enter cmd and press OK
navigate to the gPodder binary directory (C:\Program Files (x86)\gPodder\bin for the installer version, gpodder-portable-x.y.z\data\bin for the portable version)
drag and drop gpodder-cmd.exe from the windows explorer to the console window
add a space and -v to the text and press the Enter key

**Comment by cnleo** [comment]
@chlorophyll-zz thanks for your advice.

Deleting the gPodder folder did not help to work correctly with paged feeds/pagination, but to prevent the timeout in the console (timeouts not included below because only once feed is exist). ;D

(I tried the mp4 feed and the opus feed. Both have similar result. Both not working with paged feeds. Below is the opus output.)

```
C:\Program Files (x86)\gPodder\bin>"C:\Program Files (x86)\gPodder\bin\gpodder-c
md.exe" -v
Calling SetDllDirectoryW(C:\Program Files (x86)\gPodder\bin)
Successfully SetDllDirectoryW
D: windows build; registry home = F:\Podcasts\gPodder
1565016321.012149 [gpodder.log] DEBUG: ==== gPodder starts up (ui=gtk) ===
1565016321.075153 [gpodder.util] WARNING: html5lib not found, falling back to HT
MLParser
1565016321.420172 [gpodder.sync] WARNING: Could not find gpod
1565016321.421172 [gpodder.sync] WARNING: Could not find eyed3.mp3
1565016321.508178 [gpodder.gtkui.shownotes] INFO: No WebKit2 gobject bindings, s
o no HTML shownotes
1565016321.681187 [gpodder.extensions] DEBUG: Found extension "audio_converter"
in C:\Program Files (x86)\gPodder\share\gpodder\extensions\audio_converter.py
1565016321.683187 [gpodder.extensions] DEBUG: Found extension "command_on_downlo
ad" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\command_on_downlo
ad.py
1565016321.683187 [gpodder.extensions] DEBUG: Found extension "concatenate_video
s" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\concatenate_videos
.py
1565016321.684187 [gpodder.extensions] DEBUG: Found extension "enqueue_in_mediap
layer" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\enqueue_in_med
iaplayer.py
1565016321.685188 [gpodder.extensions] DEBUG: Found extension "episode_website_c
ontext_menu" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\episode_
website_context_menu.py
1565016321.686188 [gpodder.extensions] DEBUG: Found extension "gtk_statusicon" i
n C:\Program Files (x86)\gPodder\share\gpodder\extensions\gtk_statusicon.py
1565016321.687188 [gpodder.extensions] DEBUG: Found extension "minimize_on_start
" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\minimize_on_start.p
y
1565016321.687188 [gpodder.extensions] DEBUG: Found extension "mpris-listener" i
n C:\Program Files (x86)\gPodder\share\gpodder\extensions\mpris-listener.py
1565016321.688188 [gpodder.extensions] DEBUG: Found extension "normalize_audio"
in C:\Program Files (x86)\gPodder\share\gpodder\extensions\normalize_audio.py
1565016321.689188 [gpodder.extensions] DEBUG: Found extension "notification" in
C:\Program Files (x86)\gPodder\share\gpodder\extensions\notification.py
1565016321.691188 [notification] INFO: Could not find PyNotify.
1565016321.691188 [gpodder.extensions] INFO: Module loaded: C:\Program Files (x8
6)\gPodder\share\gpodder\extensions\notification.py
1565016321.692188 [gpodder.extensions] DEBUG: Found extension "notification-win3
2" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\notification-win32
.py
1565016321.695188 [gpodder.extensions] INFO: Module loaded: C:\Program Files (x8
6)\gPodder\share\gpodder\extensions\notification-win32.py
1565016321.696188 [gpodder.extensions] DEBUG: Found extension "rename_download"
in C:\Program Files (x86)\gPodder\share\gpodder\extensions\rename_download.py
1565016321.697188 [gpodder.extensions] DEBUG: Found extension "rm_ogg_cover" in
C:\Program Files (x86)\gPodder\share\gpodder\extensions\rm_ogg_cover.py
1565016321.697188 [gpodder.extensions] DEBUG: Found extension "rockbox_convert2m
p4" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\rockbox_convert2m
p4.py
1565016321.698188 [gpodder.extensions] DEBUG: Found extension "rockbox_coverart"
 in C:\Program Files (x86)\gPodder\share\gpodder\extensions\rockbox_coverart.py
1565016321.699188 [gpodder.extensions] DEBUG: Found extension "sonos" in C:\Prog
ram Files (x86)\gPodder\share\gpodder\extensions\sonos.py
1565016321.700188 [gpodder.extensions] DEBUG: Found extension "tagging" in C:\Pr
ogram Files (x86)\gPodder\share\gpodder\extensions\tagging.py
1565016321.701188 [gpodder.extensions] DEBUG: Found extension "taskbar_progress"
 in C:\Program Files (x86)\gPodder\share\gpodder\extensions\taskbar_progress.py
1565016321.701188 [gpodder.extensions] DEBUG: Found extension "ted_subtitles" in
 C:\Program Files (x86)\gPodder\share\gpodder\extensions\ted_subtitles.py
1565016321.702188 [gpodder.extensions] DEBUG: Found extension "ubuntu_appindicat
or" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\ubuntu_appindicat
or.py
1565016321.704189 [gpodder.extensions] DEBUG: Found extension "ubuntu_unity" in
C:\Program Files (x86)\gPodder\share\gpodder\extensions\ubuntu_unity.py
1565016321.704189 [gpodder.extensions] DEBUG: Found extension "update_feeds_on_s
tartup" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\update_feeds_
on_startup.py
1565016321.705189 [gpodder.extensions] DEBUG: Found extension "video_converter"
in C:\Program Files (x86)\gPodder\share\gpodder\extensions\video_converter.py
1565016322.313223 [gpodder.dbsqlite] INFO: Loading podcasts
1565016322.314224 [gpodder.dbsqlite] DEBUG: Database opened.
1565016322.315223 [gpodder.dbsqlite] INFO: Loading episodes for podcast 1
1565016322.360226 [gpodder.my] INFO: Reloading settings.
1565016322.361226 [gpodder.util] DEBUG: run_in_background: <bound method UserApp
sReader.read of <gpodder.gtkui.desktopfile.UserAppsReader object at 0x04991ed0>>
 (False)
1565016322.362226 [gpodder.gtkui.desktopfile] WARNING: Parse HKEY error: HKEY_CL
ASSES_ROOT\Winamp.File\shell\Play\command ([WinError 2] Das System kann die ange
gebene Datei nicht finden)
1565016322.363226 [gpodder.gtkui.desktopfile] WARNING: Parse HKEY error: HKEY_CL
ASSES_ROOT\WMP11.AssocFile.MP3\shell\open\command ([WinError 2] Das System kann
die angegebene Datei nicht finden)
1565016322.364226 [gpodder.gtkui.desktopfile] WARNING: Parse HKEY error: HKEY_CL
ASSES_ROOT\QuickTime.mp3\shell\open\command ([WinError 2] Das System kann die an
gegebene Datei nicht finden)
1565016322.433230 [gpodder.util] DEBUG: run_in_background: <bound method gPodder
.find_partial_downloads of <gpodder.gtkui.main.gPodder object at 0x049dd3d0>> (F
alse)
1565016322.434230 [gpodder.gtkui.main] DEBUG: Setting up auto update timer with
interval 20.
1565016322.435230 [gpodder.config] DEBUG: software_update.last_check: 0 -> 15650
16322
1565016322.436230 [gpodder.util] DEBUG: run_in_background: <bound method Config.
save_thread_proc of <gpodder.gtkui.config.UIConfig object at 0x049d4d30>> (True)

1565016358.355285 [gpodder.util] DEBUG: run_in_background: <function gPodder.upd
ate_feed_cache.<locals>.update_feed_cache_proc at 0x0498cf18> (False)
1565016358.368286 [gpodder.gtkui.main] INFO: Aktualisiert CRE: Technik, Kultur,
Gesellschaft (1/1)
1565016358.607299 [gpodder.dbsqlite] DEBUG: Commit.
1565016358.610299 [gpodder.gtkui.services] DEBUG: cover download request for htt
p://cre.fm/feed/opus
1565016358.610299 [gpodder.util] DEBUG: run_in_background: <function CoverDownlo
ader.request_cover.<locals>.<lambda> at 0x02dc6810> (False)
1565016358.612299 [gpodder.gtkui.main] DEBUG: Updated channel is active, updatin
g UI
1565016358.652302 [gpodder.my] DEBUG: Processing received episode actions...
1565016358.659302 [gpodder.my] DEBUG: Received episode actions processed.
1565016358.659302 [gpodder.dbsqlite] DEBUG: Commit.
1565016358.903316 [gpodder.gtkui.model] INFO: Clearing cover from cache: http://
cre.fm/feed/opus
1565016382.437662 [gpodder.config] INFO: Flushing settings to disk
```

**Comment by chlorophyll-zz** [comment]
@cnleo Thank you too. 
> 565016321.075153 [gpodder.util] WARNING: html5lib not found, falling back to HTMLParser
> [...]
> 1565016321.508178 [gpodder.gtkui.shownotes] INFO: No WebKit2 gobject bindings, s
> o no HTML shownotes
These are the only warnings related to gpodder.
Do you have working html Shownotes with links?

I found your Issue applies to all official Releases of Version 3.10.9 . 
The missing paged Feeds is not an Error and not on your side. The official Release that you have installed is at commit 711ed276df713fde8e6f31d4bd3d2a71a7271905 , the paged feeds was one commit later 034e3ffa2b60925069d738b050f569b2514dedbd . 

So it seems, you would have to wait for the next Release after 3.10.9 to have paged feeds in the official Release, or build gpodder from source for yourself until then. Excuse me, please. I missed the fact that the official release was before the paged feeds commit. I concluded wrongly that you builded gpodder from source. Do you want me in excuse to guide you through the gpodder.net synchronization to restore your feeds?

**Comment by cnleo** [comment]
@chlorophyll-zz I have installed the appveyor installer version of gpodder 1.0.193 and paged feeds  working now (CRE starts with episode 20).

Now the right click menu entry “website” on episodes/podcasts is gone.

**OT/OOT:**
Winamp, Quicktime and WMP(11) they’ve all been gone for a long time and all audio and video files are associated with VLC (120 of 125, exception for .iso, .zip, .rar, .vlt, .wst).
https://support.microsoft.com/help/18539/windows-7-change-default-programs

I could delete the broken entry’s from windows registry, if i would find them.
There is no Winamp entry (_HKEY_CLASSES_ROOT\Winamp.File\shell\Play\command_).
Similar to QuickTime and WMP.

About .mp3 association, here are the entry’s 
cmd: `REG EXPORT HKCR\.MP3 C:\exports\mp3.txt`

> 
> [HKEY_CLASSES_ROOT\.MP3]
> @="VLC.mp3"
> "PerceivedType"="audio"
> "VLC.backup"="WMP11.AssocFile.MP3"
> 
> [HKEY_CLASSES_ROOT\.MP3\OpenWithProgIds]
> 
> [HKEY_CLASSES_ROOT\.MP3\PersistentHandler]
> @="{098f2470-bae0-11cd-b579-08002b30bfeb}"
> 
> [HKEY_CLASSES_ROOT\.MP3\shellex]
> 
> [HKEY_CLASSES_ROOT\.MP3\shellex\{BB2E617C-0920-11D1-9A0B-00C04FC2D6C1}]
> @="{9DBD2C50-62AD-11D0-B806-00C04FD706EC}"
> 
> [HKEY_CLASSES_ROOT\.MP3\shellex\{e357fccd-a995-4576-b01f-234630154e96}]
> @="{9DBD2C50-62AD-11D0-B806-00C04FD706EC}"
>

**Comment by elelay** [comment]
@chlorophyll-zz thanks for sorting this out! Indeed, I shall make a new release soon.

@cnleo the *Open website* menu item is an extension. Enable it from Preferences > Extensions > User Interface

**Comment by cnleo** [comment]
@elelay 
> Open website menu item is an extension. Enable it from Preferences > Extensions > User Interface

Oops, I completely forgot about that, thanks.

**Comment by chlorophyll-zz** [comment]
@elelay thank you for releasing autobuilds so i could point @cnleo to the unofficial release for testing.
we have tested and confirmed working paged feeds. so i would vote for closed issue. what do you think?

**Comment by elelay** [comment]
I'll close the issue after next release.
Thanks,

**Comment by cnleo** [comment]
I am a litle bit surprised. I have installed the Update 3.10.11 (2019-09-29).
The pagination still not working. Is it for 3.11?

Tested with:
https://cre.fm/feed/opus/
Redirect: https://feeds.metaebene.me/cre/opus

**Comment by chlorophyll-zz** [comment]
@cnleo testing with 3.10.11-portable. Logs follow.

**Comment by cnleo** [comment]
@chlorophyll-zz 

> ... i see 20 to 219 (from bottom to top).

Last one (or the first episodes):
https://cre.fm/feed/opus?paged=4
https://cre.fm/feed/m4a?paged=4
Strange enough there is episode 1 (CRE001) to 20 (CRE020) up to 39 in there.
But I remember I saw 20 to 219 too in the appveyor version.

Now I see 160 to 219. What exactly the paged 1 is https://cre.fm/feed/opus?paged=1
The log from the installed version:

> 2019-11-02 23:36:38,829 [gpodder.log] DEBUG: ==== gPodder starts up (ui=gtk) ===
> 2019-11-02 23:36:40,609 [gpodder.util] WARNING: html5lib not found, falling back to HTMLParser
> 2019-11-02 23:36:41,166 [gpodder.registry] DEBUG: Registering download_url resolver: <function youtube_real_download_url at 0x030c9bfc>
> 2019-11-02 23:36:41,216 [gpodder.registry] DEBUG: Registering download_url resolver: <function escapist_real_download_url at 0x030d32fc>
> 2019-11-02 23:36:41,275 [gpodder.registry] DEBUG: Registering download_url resolver: <function vimeo_real_download_url at 0x030c7fa4>
> 2019-11-02 23:36:44,219 [gpodder.sync] WARNING: Could not find gpod
> 2019-11-02 23:36:44,220 [gpodder.sync] WARNING: Could not find eyed3.mp3
> 2019-11-02 23:36:44,514 [gpodder.gtkui.shownotes] INFO: No WebKit2 gobject bindings, so no HTML shownotes
> 2019-11-02 23:36:45,540 [gpodder.extensions] DEBUG: Found extension "audio_converter" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\audio_converter.py
> 2019-11-02 23:36:45,584 [gpodder.extensions] DEBUG: Found extension "command_on_download" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\command_on_download.py
> 2019-11-02 23:36:45,627 [gpodder.extensions] DEBUG: Found extension "concatenate_videos" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\concatenate_videos.py
> 2019-11-02 23:36:45,655 [gpodder.extensions] DEBUG: Found extension "enqueue_in_mediaplayer" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\enqueue_in_mediaplayer.py
> 2019-11-02 23:36:45,730 [enqueue_in_mediaplayer] WARNING: Win32 player not found: HKEY_CLASSES_ROOT\Winamp.File\shell\Enqueue\command ([WinError 2] Das System kann die angegebene Datei nicht finden)
> 2019-11-02 23:36:45,731 [gpodder.extensions] INFO: Module loaded: C:\Program Files (x86)\gPodder\share\gpodder\extensions\enqueue_in_mediaplayer.py
> 2019-11-02 23:36:45,732 [gpodder.extensions] DEBUG: Found extension "episode_website_context_menu" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\episode_website_context_menu.py
> 2019-11-02 23:36:45,742 [gpodder.extensions] INFO: Module loaded: C:\Program Files (x86)\gPodder\share\gpodder\extensions\episode_website_context_menu.py
> 2019-11-02 23:36:45,742 [gpodder.extensions] DEBUG: Found extension "gtk_statusicon" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\gtk_statusicon.py
> 2019-11-02 23:36:45,757 [gpodder.extensions] DEBUG: Found extension "minimize_on_start" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\minimize_on_start.py
> 2019-11-02 23:36:45,771 [gpodder.extensions] DEBUG: Found extension "mpris-listener" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\mpris-listener.py
> 2019-11-02 23:36:45,790 [gpodder.extensions] DEBUG: Found extension "normalize_audio" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\normalize_audio.py
> 2019-11-02 23:36:45,812 [gpodder.extensions] DEBUG: Found extension "notification" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\notification.py
> 2019-11-02 23:36:45,821 [notification] INFO: Could not find PyNotify.
> 2019-11-02 23:36:45,821 [gpodder.extensions] INFO: Module loaded: C:\Program Files (x86)\gPodder\share\gpodder\extensions\notification.py
> 2019-11-02 23:36:45,822 [gpodder.extensions] DEBUG: Found extension "notification-win32" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\notification-win32.py
> 2019-11-02 23:36:45,860 [gpodder.extensions] INFO: Module loaded: C:\Program Files (x86)\gPodder\share\gpodder\extensions\notification-win32.py
> 2019-11-02 23:36:45,861 [gpodder.extensions] DEBUG: Found extension "rename_download" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\rename_download.py
> 2019-11-02 23:36:45,923 [gpodder.extensions] DEBUG: Found extension "rm_ogg_cover" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\rm_ogg_cover.py
> 2019-11-02 23:36:46,000 [gpodder.extensions] DEBUG: Found extension "rockbox_convert2mp4" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\rockbox_convert2mp4.py
> 2019-11-02 23:36:46,043 [gpodder.extensions] DEBUG: Found extension "rockbox_coverart" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\rockbox_coverart.py
> 2019-11-02 23:36:46,055 [gpodder.extensions] DEBUG: Found extension "sonos" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\sonos.py
> 2019-11-02 23:36:46,070 [gpodder.extensions] DEBUG: Found extension "tagging" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\tagging.py
> 2019-11-02 23:36:46,092 [gpodder.extensions] DEBUG: Found extension "taskbar_progress" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\taskbar_progress.py
> 2019-11-02 23:36:46,104 [gpodder.extensions] DEBUG: Found extension "ted_subtitles" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\ted_subtitles.py
> 2019-11-02 23:36:46,122 [gpodder.extensions] DEBUG: Found extension "ubuntu_appindicator" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\ubuntu_appindicator.py
> 2019-11-02 23:36:46,133 [gpodder.extensions] DEBUG: Found extension "ubuntu_unity" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\ubuntu_unity.py
> 2019-11-02 23:36:46,158 [gpodder.extensions] DEBUG: Found extension "update_feeds_on_startup" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\update_feeds_on_startup.py
> 2019-11-02 23:36:46,177 [gpodder.extensions] DEBUG: Found extension "video_converter" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\video_converter.py
> 2019-11-02 23:36:46,196 [gpodder.extensions] DEBUG: Found extension "youtube-dl" in C:\Program Files (x86)\gPodder\share\gpodder\extensions\youtube-dl.py
> 2019-11-02 23:36:46,263 [gpodder.registry] DEBUG: Registering feed_handler resolver: <bound method SoundcloudFeed.fetch_channel of <class 'gpodder.plugins.soundcloud.SoundcloudFeed'>>
> 2019-11-02 23:36:46,263 [gpodder.registry] DEBUG: Registering feed_handler resolver: <bound method SoundcloudFeed.fetch_channel of <class 'gpodder.plugins.soundcloud.SoundcloudFavFeed'>>
> 2019-11-02 23:36:55,286 [gpodder.dbsqlite] INFO: Loading podcasts
> 2019-11-02 23:36:55,295 [gpodder.dbsqlite] DEBUG: Database opened.
> 2019-11-02 23:36:55,296 [gpodder.dbsqlite] INFO: Loading episodes for podcast 4
> 2019-11-02 23:36:55,380 [gpodder.dbsqlite] INFO: Loading episodes for podcast 5
> [...]
> 2019-11-02 23:36:56,902 [gpodder.dbsqlite] INFO: Loading episodes for podcast 259
> 2019-11-02 23:36:56,909 [gpodder.dbsqlite] INFO: Loading episodes for podcast 260
> 2019-11-02 23:36:57,076 [gpodder.my] INFO: Reloading settings.
> 2019-11-02 23:36:57,077 [gpodder.util] DEBUG: run_in_background: <bound method UserAppsReader.read of <gpodder.gtkui.desktopfile.UserAppsReader object at 0x10cbde8c>> (False)
> 2019-11-02 23:36:57,078 [gpodder.gtkui.desktopfile] WARNING: Parse HKEY error: HKEY_CLASSES_ROOT\Winamp.File\shell\Play\command ([WinError 2] Das System kann die angegebene Datei nicht finden)
> 2019-11-02 23:36:57,079 [gpodder.gtkui.desktopfile] WARNING: Parse HKEY error: HKEY_CLASSES_ROOT\WMP11.AssocFile.MP3\shell\open\command ([WinError 2] Das System kann die angegebene Datei nicht finden)
> 2019-11-02 23:36:57,080 [gpodder.gtkui.desktopfile] WARNING: Parse HKEY error: HKEY_CLASSES_ROOT\QuickTime.mp3\shell\open\command ([WinError 2] Das System kann die angegebene Datei nicht finden)
> 2019-11-02 23:37:01,126 [gpodder.gtkui.model] DEBUG: cached thumb wrong size: (29, 29) != 30
> 2019-11-02 23:37:30,075 [gpodder.util] DEBUG: run_in_background: <bound method gPodder.find_partial_downloads of <gpodder.gtkui.main.gPodder object at 0x040f4a2c>> (False)
> 2019-11-02 23:37:39,600 [gpodder.gtkui.main] DEBUG: find_partial_downloads done, calling extensions
> 2019-11-02 23:38:22,150 [gpodder.util] DEBUG: run_in_background: <function gPodder.update_feed_cache.<locals>.update_feed_cache_proc at 0x04fc8224> (False)
> 2019-11-02 23:38:22,174 [gpodder.gtkui.main] INFO: Aktualisiert CRE: Technik, Kultur, Gesellschaft (1/1)
> 2019-11-02 23:38:22,776 [gpodder.dbsqlite] DEBUG: Commit.
> 2019-11-02 23:38:23,026 [gpodder.gtkui.services] DEBUG: cover download request for http://cre.fm/feed/opus/
> 2019-11-02 23:38:23,027 [gpodder.util] DEBUG: run_in_background: <function CoverDownloader.request_cover.<locals>.<lambda> at 0x04fc8c44> (False)
> 2019-11-02 23:38:23,300 [gpodder.gtkui.main] DEBUG: Updated channel is active, updating UI
> 2019-11-02 23:38:23,304 [gpodder.my] DEBUG: Processing received episode actions...
> 2019-11-02 23:38:23,448 [gpodder.my] DEBUG: Received episode actions processed.
> 2019-11-02 23:38:23,450 [gpodder.dbsqlite] DEBUG: Commit.
> 2019-11-02 23:38:23,655 [gpodder.gtkui.model] INFO: Clearing cover from cache: http://cre.fm/feed/opus/
> 2019-11-02 23:41:43,237 [gpodder.util] DEBUG: run_in_background: <bound method Config.save_thread_proc of <gpodder.gtkui.config.UIConfig object at 0x040b6f8c>> (True)
> 2019-11-02 23:42:43,249 [gpodder.config] INFO: Flushing settings to disk

**Comment by chlorophyll-zz** [comment]
@cnleo 
i think you have a few options: 
you could test with the portable release. then we could make a conlusion wether the problem with paged feed comes from the installation and could exclude a problem from the config. 

for that, you would export your feeds in your "stationary" install from the menu abonements -> export to opml, and import the opml in the portable version from the menu abonements -> import from opml. this would be a bit of a hassle for you with a second portable installation, but in the interest of your specific problem you could test further from what condition stems the problem. 

what you could also do is, what i would consider a "long range resolution", you export the opml file from the abonements menu, and take the steps before the appveyor test install:
rename the gpodder folder in "my documents" to save your downloaded pods (if you want that) 
and reinstall "stationary" gpodder. with "stationary" i mean the not-portable installation in the "program files" folder.

**Comment by cnleo** [comment]
OK, after many experimental and uninstalls and installs the solution is a little… out of scope.

Is your podcast (or here the CRT) for a time period not updated, gpodder will not try to download the feed AND the “new” paged feeds. Sure there is nothing new ;D

Change the entries _“http_last_modified”_ or/and may in _“http_etag”_ in Database SQLite file.

SQL statement:  
`UPDATE "main"."podcast" SET "http_last_modified" = NULL;`

After that, start gpodder and look for new “old” episodes.

Special for CRE:
After I changed the entry in Database for CRE, the episodes was all loaded from 1 to 219.
Beware, in this case, episode 1 to 20 are only visible if the default of watch 200 episodes per podcast is changed.

Conclusion:
Is there a need for a new check for already parsed **paged** feeds?
The “problem” occurs in the update environment.
Fresh or clean Database for install and portable version will not have this problem.

**Comment by thp** [comment]
Could be made so that on update (just the next one or any version update in particular), `http_last_modified` and `http_etag` are both set to `NULL` for each podcast in the database (note that `http_etag` also needs to be set to `NULL` in general, since [`If-None-Match`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-None-Match) alone can also return HTTP 304 (resulting in the same problem) -- `http_last_modified` changes [`If-Modified-Since`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-Modified-Since) behavior.

In that particular case, it might be that the server didn't support `ETag`/`If-None-Match`, so just clearing the last modified field (client does not send `If-Modified-Since` header) was enough to force an update.

Also note that if/when the podcast would have been updated properly, it would also have caused the paged feeds to become visible.

**Comment by cnleo** [comment]
A problem with the solution (all feeds reload and parse) is, for user they updated there podcasts automatically and there are many podcasts in there list, it has a huge performance problem with “program/application without responds/reacts” message.
I had some problems with the load of all podcast (261) from OPML in a clean gPodder install.
But that’s possible an other issue.

**Comment by cnleo** [comment]
There is another problem with old Podcast feeds in an old list/database and forcing a update with `http_last_modified `and `http_etag` set to `NULL`.
If the feed is no more available. And you get a 404 response. The podcast will vanish its content in gPodder; still if you selected “archived” before. The audio/video files are still in there download folder.

**Comment by chlorophyll-zz** [comment]
@cnleo much edge cases :-p
for the vanishing, i think i read that in a very far away issue as well.
